src/clients/SpokePoolClient/EVMSpokePoolClient.ts

```diff
@@ -0,0 +1,234 @@
+import { Contract, EventFilter } from "ethers";
+import {
+  BigNumber,
+  DepositSearchResult,
+  getDepositIdAtBlock,
+  getNetworkName,
+  InvalidFill,
+  MakeOptional,
+  toBN,
+} from "../../utils";
+import {
+  EventSearchConfig,
+  paginatedEventQuery,
+  sortEventsAscendingInPlace,
+  spreadEventWithBlockNumber,
+} from "../../utils/EventUtils";
+import { isUpdateFailureReason } from "../BaseAbstractClient";
+import { knownEventNames, SpokePoolClient, SpokePoolUpdate } from "./SpokePoolClient";
+import winston from "winston";
+import { HubPoolClient } from "../HubPoolClient";
+import { SpokePool } from "../../typechain";
+import {
+  findDepositBlock,
+  getMaxFillDeadlineInRange as getMaxFillDeadline,
+  getTimeAt as _getTimeAt,
+  relayFillStatus,
+  isZeroAddress,
+} from "../../utils/SpokeUtils";
+import { DepositWithBlock, FillStatus, RelayData } from "../../interfaces";
+
+/**
+ * An EVM-specific SpokePoolClient.
+ */
+export class EVMSpokePoolClient extends SpokePoolClient {
+  constructor(
+    logger: winston.Logger,
+    public readonly spokePool: Contract,
+    hubPoolClient: HubPoolClient | null,
+    chainId: number,
+    deploymentBlock: number,
+    eventSearchConfig: MakeOptional<EventSearchConfig, "toBlock"> = { fromBlock: 0, maxBlockLookBack: 0 }
+  ) {
+    super(logger, hubPoolClient, chainId, deploymentBlock, eventSearchConfig);
+  }
+
+  public async getTimestampForBlock(blockTag: number): Promise<number> {
+    const block = await this.spokePool.provider.getBlock(blockTag);
+    return Number(block.timestamp);
+  }
+
+  public relayFillStatus(
+    relayData: RelayData,
+    blockTag?: number | "latest",
+    destinationChainId?: number
+  ): Promise<FillStatus> {
+    return relayFillStatus(this.spokePool, relayData, blockTag, destinationChainId);
+  }
+
+  /**
+   * Finds the deposit id at a specific block number.
+   * @param blockTag The block number to search for the deposit ID at.
+   * @returns The deposit ID.
+   */
+  public override _getDepositIdAtBlock(blockTag: number): Promise<BigNumber> {
+    return getDepositIdAtBlock(this.spokePool as SpokePool, blockTag);
+  }
```

Comment:
fwiw, I'm propsing to remove this function in [this PR](https://github.com/across-protocol/sdk/pull/926/files) because I don't think we actually need it.

```suggestion
```

Reply:
https://github.com/across-protocol/sdk/pull/935